cmd/gateway/zerostor/errors/warn.go

```diff
@@ -0,0 +1,38 @@
+package errors
+
+type causer interface {
+	Cause() error
+}
+
+//Warning is a wrapper that demotes an error to warning level
+type warning struct {
+	error
+}
+
+//Cause gets the cause of the warning
+func (w warning) Cause() error {
+	return w.error
+}
+
+//Warning marks this error as a warning
+func Warning(err error) error {
+	if _, ok := err.(warning); ok {
+		return err
+	}
+
+	return warning{err}
+}
+
+//IsWarning checks if this error is a warning
+func IsWarning(err error) bool {
+	for {
+		switch cause := err.(type) {
+		case warning:
+			return true
+		case causer:
+			err = cause.Cause()
```

Comment:
why do you assign err ?
This function should just check the type of the error, not modifying it

Reply:
this won't change the passed error. I am checking if this is a warning, or caused by a warning. so I am checking if it has a causer that is a warning as well

Reply:
I didn't notice this was in a loop, ok got it